ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -211,8 +211,11 @@ let infer_vgpu_map ~__context ?remote vm =
     List.map (fun self ->
       let vgpu = Db.VGPU.get_record ~__context ~self in
       let device = vgpu.API.vGPU_device in
+      let pgpu = vgpu.API.vGPU_scheduled_to_be_resident_on in
+      if pgpu = Ref.null then (* the VM was halted/suspended *)
+        raise (Xenops_interface.Does_not_exist("PGPU", Ref.string_of vm));
```

Comment:
and I don't like the idea of raising Xenops_interface errors - that's quite misleading.

Reply:
Also, it'd be good to have a comment here saying precisely _why_ we're handling this carefully.